libraries/lib-stretching-sequence/AudioSegmentFactory.h

```diff
@@ -0,0 +1,40 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSegmentFactory.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+
+#pragma once
+
+#include "AudioSegmentFactoryInterface.h"
+
+class ClipInterface;
+using ClipConstHolders = std::vector<std::shared_ptr<const ClipInterface>>;
+
+class STRETCHING_SEQUENCE_API AudioSegmentFactory final :
+    public AudioSegmentFactoryInterface
+{
+public:
+   AudioSegmentFactory(
+      size_t sampleRate, size_t numChannels, const ClipConstHolders& clips);
```

Comment:
`size_t` seems a strange choice of type to describe a sample rate.  It's not sizing something in memory.

Reply:
Never thought of `size_t` that way ... Will use just int.